src/Forms/TimeField_Readonly.php

```diff
@@ -14,12 +14,10 @@ class TimeField_Readonly extends TimeField
 
     public function Field($properties = array())
     {
-        if ($this->valueObj) {
-            $val = Convert::raw2xml($this->valueObj->toString($this->getConfig('timeformat')));
-        } else {
-            // TODO Localization
-            $val = '<i>(not set)</i>';
-        }
+        //Allow the users locale and the field to define the format
+        $this->setHTML5(false);
```

Comment:
Could you help me understand why we need to set this to false? My (limited) understanding is that setting this to true would mean that browser would handle setting the date formatting to the correct locale

Reply:
TimeField_Readonly uses a span to render, setting this to false makes `TimeField::Value()` return it's value based on the user's cms preference rather than using the ISO format.

Reply:
OK, so what you're saying is that:
- a) if you're logged, and you've set a CMS preference for date formatting, then that will display
- b) if you're not logged in, it will render based on whatever default is defined in the CMS (seems like this may causes may cause confusion for visitors from a different continent from where the company is hosted i.e. seeing d/m/y rendered as m/d/y)
?

Reply:
Yes though this is only a time not a date, which to my limited localization understanding is there's really 2 dominate formats 12hr and 24hr. The only way to solve this I think would be to use a disabled or readonly `<input type="time"/>` to allow the browser to format properly.

Reply:
Yeah seems like we might need to do something like that

I'm not too keen on changing the readonly version html5 to false while the default version is true https://github.com/silverstripe/silverstripe-framework/blob/4/src/Forms/TimeField.php#L68 - seems like if you had a writable timefield and and readonly timefield field in the same form then their formatting may be different

Would it be worth removing the `$this->setHTML5(false);`, as well as changes in TimeField->performReadonlyTransformation(), and changing the `<span ...` below to `<input type="time" disabled="disabled" class=\"readonly\" id=\"" . $this->ID() . "\">$val</span>";`

Let me know if I'm interpreting this incorrectly

Reply:
There I switched it back to using the template to render which uses the input type time and I also disabled it. That seems to solve the original issue and prevent the field from being saved into a data object.

Reply:
OK seems good, though I found an issue during testing - It's triggered by the ->performReadonlyTransformation()

Before applying the patch, I was getting the (not set) issue

![image](https://user-images.githubusercontent.com/4809037/73988392-72a4f200-49a7-11ea-84f7-1861790f08f2.png)

Reply:
Seems fixed now let me know if the issue still occurs it wasn't for me, apparently `TimeField::getTimeLength()` returns something other than what is stored in `TimeField::$timeLength`. So I'm now just using the raw value in the property instead when creating the copy. I also tweaked setting the locale for the same reason.